flink-streaming-connectors/flink-connector-nifi/src/main/java/org/apache/flink/streaming/connectors/nifi/NiFiSource.java

```diff
@@ -143,4 +144,13 @@ public void close() throws Exception {
 		client.close();
 	}
 
+ /**
+	* {@inheritDoc}
+	* <p>
+	* Sets the {@link #isRunning} flag to {@code false}.
+	*/
+	@Override
+	public void stop() {
+		this.isRunning = false;
```

Comment:
I guess we need to for now, until stop() is wired into the StreamGraph i.e. Thanks for pointing that out.